cnf-tests/testsuites/e2esuite/nmstate/nmstate_sriov.go

```diff
@@ -0,0 +1,207 @@
+
+	BeforeEach(func() {
+		if !networks.IsSriovOperatorInstalled() {
+			Skip("SR-IOV operator not installed on the cluster.")
+		}
+
+		err := namespaces.Create(testNamespace, client.Client)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("CleanSriov...")
+		networks.CleanSriov(sriovclient)
+
+		By("Discover SRIOV devices")
+		sriovInfos, err = cluster.DiscoverSriov(sriovclient, namespaces.SRIOVOperator)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(len(sriovInfos.Nodes)).ToNot(BeZero(), "no node with SR-IOV NICs found")
+
+		nmstate := nmstatev1.NMState{ObjectMeta: metav1.ObjectMeta{Name: "nmstate"}}
+		err = client.Client.Create(context.Background(), &nmstate)
+		Expect(err).ToNot(HaveOccurred())
+		DeferCleanup(client.Client.Delete, context.Background(), &nmstate)
+	})
+
+	Context("when a SriovNetworkNodePolicy and NodeNetworkConfigurationPolicySpec target the same NIC", func() {
+
+		BeforeEach(func() {
+			namespaces.CleanPods(testNamespace, client.Client)
+		})
+
+		It("VFs should spawn correctly", func() {
+			node, devices, err := findNonPrimarySriovDevicesAndNode(sriovInfos)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(devices)).ToNot(BeZero())
+
+			By("Using node " + node)
+
+			testDevice := devices[0]
+			By("Using device " + testDevice.Name)
+
+			By("Configuring an IPv4 address on Physical Function via NMState")
+			nodeNetworkConfigPolicy := nmstatev1.NodeNetworkConfigurationPolicy{
+				ObjectMeta: metav1.ObjectMeta{Name: "test-sriov-device-" + testDevice.Name},
+				Spec: nmstateshared.NodeNetworkConfigurationPolicySpec{
+					NodeSelector: map[string]string{
+						"kubernetes.io/hostname": node,
+					},
+					DesiredState: nmstateshared.State{
+						Raw: nmstateshared.RawState(fmt.Sprintf(nmstateDesiredStateTemplateStaticIP, testDevice.Name)),
+					},
+				},
+			}
+			err = client.Client.Create(context.Background(), &nodeNetworkConfigPolicy)
+			Expect(err).ToNot(HaveOccurred())
+			DeferCleanup(client.Client.Delete, context.Background(), &nodeNetworkConfigPolicy)
+
+			waitForNMStatePolicyToBeStable(nodeNetworkConfigPolicy.Name)
+
+			By("Configuring VF on device via SriovNetworkNodePolicy")
+			sriovNetworkNodePolicy, err := sriovnetwork.CreateSriovPolicy(sriovclient, "test-nmstate-", namespaces.SRIOVOperator, testDevice.Name, node, 10, "testnmstateresource", "netdevice")
+			Expect(err).ToNot(HaveOccurred())
+			DeferCleanup(sriovclient.Delete, context.Background(), sriovNetworkNodePolicy)
+
+			Eventually(func() int64 {
+				testedNode, err := sriovclient.Nodes().Get(context.Background(), node, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				resNum := testedNode.Status.Allocatable[corev1.ResourceName("openshift.io/testnmstateresource")]
+				capacity, _ := resNum.AsInt64()
+				return capacity
+			}).
+				WithPolling(5*time.Second).
+				WithTimeout(10*time.Minute).
+				Should(Equal(int64(10)), "the number of the observed VFs is not that same as the one sriov created")
```

Comment:
Wdyt about going deeper in checking that both operators did their part? I would check the ip on the interface on the host, to make sure (k)nmstate did its part and probably make some tests on the vf.

Reply:
sounds good! changing